pyntcloud/samplers/points.py

```diff
@@ -24,3 +27,42 @@ def compute(self):
         if self.n > len(self.points):
             raise ValueError("n can't be higher than the number of points in the PyntCloud.")
         return self.points.sample(self.n).reset_index(drop=True)
+
+
+class FarthestPointsSampler(PointsSampler):
+    """
+    Parameters
+    ----------
+    n: int
+        Number of unique points that will be chosen.
+    """
+
+    def __init__(self, *, pyntcloud, n, d_metric=np.diag([1., 1., 1.])):
+        """d_metric -> Euclidean distance space by default, can be modified to other Mahalanobis distance as well"""
+        super().__init__(pyntcloud=pyntcloud)
+        self.n = n
+        if not np.all(np.linalg.eigvals(d_metric) >= 0):
+            raise ValueError("the Mahalanobis matrix must be positive semidefinite")
+        self.d_metric = d_metric
+
+    def cal_distance(self, point, solution_set):
+        sum = 0
+        for pt in solution_set:
+            sum += np.dot(point[:3]-pt[:3], self.d_metric)
+        return sum
+
+    def compute(self):
```

Comment:
I think that this implementation was originally though for working with python lists. However PyntCloud is using pandas DataFrames/Numpy arrays internally.

This has 2 main implications:

- The current code won't work as it is right now due to the usage of `remaining_points.pop`, which is causing the following error:

```python
AttributeError: 'numpy.ndarray' object has no attribute 'pop'
```

- The code could be rewritten more efficient and concise using one or more of the following options:

a) numpy vectorization:
https://realpython.com/numpy-array-programming/
b) numba
https://numba.pydata.org/numba-doc/dev/user/5minguide.html
c) pandas operations
https://pandas.pydata.org/pandas-docs/stable/getting_started/basics.html